### ISSUE TITLE
chore: metadata log format is irrelevant

### ISSUE THREAD

**Issue by shreemaan-abhishek** [apache/apisix#10494]
### Description

Taking an example of Kafka Logger, the plugin metadata log_format is sourced from log_util.lua:

https://github.com/apache/apisix/blob/4aeda49d853e274289792008b4c14928ebb46a5d/apisix/plugins/kafka-logger.lua#L133C1-L139C2

https://github.com/apache/apisix/blob/4aeda49d853e274289792008b4c14928ebb46a5d/apisix/utils/log-util.lua#L33C1-L43C1

But a totally different log_format goes unused and a different log format is in use by default:

https://github.com/apache/apisix/blob/4aeda49d853e274289792008b4c14928ebb46a5d/apisix/utils/log-util.lua#L132C1-L161C1

This is not inline with the documentation:

<img width="816" alt="image" src="https://github.com/apache/apisix/assets/61597896/30226f0f-8f3c-4ede-854f-23e7fef0fff8">

https://apisix.apache.org/docs/apisix/plugins/kafka-logger/#metadata

### How this can be fixed:

- if the metadata log format is not in use (can be done by removing it and running the CI) then get rid of it.
- update the documentation to show the correct log format.
- check if other logger plugins have the same issue.

**Comment by kayx23** [comment]
You mean the default log_format of plugin metadata is not used right?

Because plugin metadata can still be used, providing that `meta_format` is set to `default`.

**Comment by luoluoyuyu** [comment]
hi @shreemaan-abhishek , I have a question in the code seems to first determine whether to use a custom meta_format, no custom meta_format will use the default log_format, my understanding is correct? https://github.com/apache/apisix/blob/master/apisix/utils/log-util.lua#L240-L253

**Comment by shreemaan-abhishek** [comment]
@kayx23 

> Because plugin metadata can still be used, providing that meta_format is set to default.

That is right according to the code but when I used `kafka-logger` without modifying the `log_format` the [default log format](https://github.com/shreemaan-abhishek/apisix/blob/105993f40f34768cd0a86035d01393346fff3a94/apisix/utils/log-util.lua#L37-L41) wasn't used instead the [full log format](https://github.com/shreemaan-abhishek/apisix/blob/105993f40f34768cd0a86035d01393346fff3a94/apisix/utils/log-util.lua#L133-L160) was used.

@Vacant2333 you might want to confirm this ⬆️. You can refer this documentation for kafka-logger: https://docs.api7.ai/hub/kafka-logger

**Comment by kayx23** [comment]
> That is right according to the code but when I used kafka-logger without modifying the log_format the default log format wasn't used instead the full log format was used.

Yes agree, that is indeed the current behaviour. Therefore I didn't mention the [default log format](https://github.com/shreemaan-abhishek/apisix/blob/105993f40f34768cd0a86035d01393346fff3a94/apisix/utils/log-util.lua#L37-L41) in the new doc as it never really kicks in. The full log format is actually the default.

**Comment by Vacant2333** [comment]
> @kayx23
> 
> > Because plugin metadata can still be used, providing that meta_format is set to default.
> 
> That is right according to the code but when I used `kafka-logger` without modifying the `log_format` the [default log format](https://github.com/shreemaan-abhishek/apisix/blob/105993f40f34768cd0a86035d01393346fff3a94/apisix/utils/log-util.lua#L37-L41) wasn't used instead the [full log format](https://github.com/shreemaan-abhishek/apisix/blob/105993f40f34768cd0a86035d01393346fff3a94/apisix/utils/log-util.lua#L133-L160) was used.
> 
> @Vacant2333 you might want to confirm this ⬆️. You can refer this documentation for kafka-logger: https://docs.api7.ai/hub/kafka-logger

yes, the documentation shows the right log for now, i will set the default log format exactly same as the full_log format, its a better wayy than remove the default log format

**Comment by Vacant2333** [comment]
I have found the same issue in other logger plugins, and some plugins use this format(https://github.com/Vacant2333/apisix/blob/94fc894889d75422ee9b64f91ec3369158d19648/apisix/plugins/file-logger.lua#L58) but do not actually support setting a default meta_format like 'default.' By default, they only output the full log

**Comment by Vacant2333** [comment]
I am currently unsure how to address this issue. Our goal is to have the logger output in the form of a full log by default. The current situation is that all loggers output in the full log format (contrary to the documentation), and the [metadata_schema_log_format](https://github.com/apache/apisix/blob/225e18f389a7593f356599999e41f77418e7732c/apisix/utils/log-util.lua#L35C26-L35C26) does not seem to take effect at all, even though almost every logger [references it](https://github.com/apache/apisix/blob/225e18f389a7593f356599999e41f77418e7732c/apisix/plugins/kafka-logger.lua#L137).

We have two implementation options:

1. Remove all unused code and output in the full log format.
2. Set the full log format as the default format, meaning to make the previously ineffective code effective and switch to the full log.

I am currently attempting the second option, which may result in some unreasonable changes:

1. The current full log includes entries [such as](https://github.com/apache/apisix/blob/225e18f389a7593f356599999e41f77418e7732c/apisix/utils/log-util.lua#L157C1-L159C40):

   - latency

   - upstream_latency

   - apisix_latency

     and similar entries not present in the apisix/nginx variables. It might be unreasonable to handle these entries separately.

2. The current full log has a [two-layer structure](https://github.com/apache/apisix/blob/225e18f389a7593f356599999e41f77418e7732c/apisix/utils/log-util.lua#L134C13-L134C13), but the logger's provided log format is [single-layered](https://github.com/apache/apisix/blob/225e18f389a7593f356599999e41f77418e7732c/apisix/utils/log-util.lua#L63C15-L63C15) (multi-layered structure has been implemented in the [PR](https://github.com/apache/apisix/pull/10588)).

In summary, I need some assistance with the above.
cc @kayx23  @shreemaan-abhishek  and maybe more

**Comment by shreemaan-abhishek** [comment]
can an extra boolean attribute like `include_latencies` be added? If `true` calculate latencies and add it to the log entry? @Vacant2333

**Comment by Vacant2333** [comment]
> can an extra boolean attribute like `include_latencies` be added? If `true` calculate latencies and add it to the log entry? @Vacant2333

I have already discussed with @moonming . We should minimize modifications and remove unused code. The default values for the meta log format of all Loggers will be removed (they are not enabled). After the modifications are completed, users must customize the meta log format; otherwise, it will not have default values. I will also update the documentation.🤔
